core/src/entrypoint/entity.rs

```diff
@@ -131,203 +124,143 @@ pub struct EntityBuilder {
     text: String,
 }
 
-impl EntityBuilder {
-    pub fn new() -> Self {
-        EntityBuilder {
-            title: String::new(),
-            text: String::new(),
-        }
-    }
+/// Extract the key-value pairs from the `Infobox` template. Returns `None` if
+/// no such template exists.
+fn extract_info(nodes: &[Node]) -> Option<BTreeMap<String, Span>> {
+    let info_parameters = nodes.iter().find_map(|node| {
+        let (name, parameters) = node.as_template()?;
+        name.iter()
+            .any(|n| n.as_text().is_some_and(|v| v.starts_with("Infobox")))
+            .then_some(parameters)
+    })?;
+    let info = info_parameters
+        .iter()
+        .filter_map(|Parameter { name, value, .. }| {
+            if value.is_empty() {
+                return None;
+            }
+            let key = name.as_ref()?.first()?.as_text()?;
+            Some((key.to_string(), Span::from(value.as_slice())))
+        })
+        .collect();
+    Some(info)
+}
 
-    pub fn append_title(&mut self, text: &str) {
-        self.title.push_str(text);
+fn extract_text(nodes: &[Node]) -> (Option<Span>, Vec<Paragraph>) {
+    fn render_template(name: &str, parameters: &[Parameter]) -> Option<String> {
```

Comment:
I would prefer not to have nested functions, even though I see why you have done it here

Reply:
This was left inside due to laziness ;) Moving to module scope.